### PR TITLE
Fix no changelog API response

### PR DESCRIPTION
**Pull request by mdegat01**
## Proposed change

The status code of the changelog API when an addon had no changelog was accidently chnaged in #4972 . This PR changes it back to a 200 in this case and adds a unit test. It also fixes the `/store/addons/{slug}/changelog` API as it appears that hasn't worked for a long time (frontend is still using the supposedly deprecated `/addons/{slug}/changelog` and that did work).

As an aside this feels very hacky. We should simply return a 404 and the frontend should have the text to display in that case with localization. Supervisor should not be returning the message presented directly to users in this case.

## Type of change

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (which adds functionality to the supervisor)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue: fixes https://github.com/home-assistant/core/issues/116470
- This PR is related to issue:
- Link to documentation pull request:
- Link to cli pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] Local tests pass. **Your PR cannot be merged unless tests pass**
- [ ] There is no commented out code in this PR.
- [ ] I have followed the [development checklist][dev-checklist]
- [ ] The code has been formatted using Ruff (`ruff format supervisor tests`)
- [ ] Tests have been added to verify that the new code works.

If API endpoints of add-on configuration are added/changed:

- [ ] Documentation added/updated for [developers.home-assistant.io][docs-repository]

[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[docs-repository]: https://github.com/home-assistant/developers.home-assistant

### CODE REVIEW

**Review by sairon — APPROVED** [review]
Thanks, it's downfall of [this](https://github.com/home-assistant/supervisor/pull/4972/files#diff-c86c86a5b85990402f5c8c7ac8897ebddb45f6217778c7269ac3d911f2057f9aL105-L107) change to be precise. I will take a look if it's the only place that relied on this rather weird handling but this PR fixes the issue, so :+1:

**Review by agners — APPROVED** [review]

File: tests/api/test_store.py [hunk 1]
```diff
@@ -188,3 +188,14 @@ async def container_events_task(*args, **kwargs):
...
     await _container_events_task
+
+
+@pytest.mark.parametrize("resource", ["store/addons", "addons"])
+async def test_api_store_addons_no_changelog(
+    api_client: TestClient, coresys: CoreSys, store_addon: AddonStore, resource: str
+):
+    """Test /store/addons/{addon}/changelog REST API."""
+    resp = await api_client.get(f"/{resource}/{store_addon.slug}/changelog")
+    assert resp.status == 200
```
Comment (sairon) [inline tests/api/test_store.py:199]: Maybe just add a comment here to explain this behavior is expected by the frontend? And change it to 404 once frontend is able to handle the errors graciously.

File: tests/api/test_store.py [hunk 2]
```diff
@@ -188,3 +188,14 @@ async def container_events_task(*args, **kwargs):
     assert resp.status == 200
 
     await _container_events_task
+
+
+@pytest.mark.parametrize("resource", ["store/addons", "addons"])
+async def test_api_store_addons_no_changelog(
+    api_client: TestClient, coresys: CoreSys, store_addon: AddonStore, resource: str
+):
+    """Test /store/addons/{addon}/changelog REST API."""
```
Comment (agners) [inline tests/api/test_store.py:197]:
```suggestion
    """Test /store/addons/{addon}/changelog REST API.
    
    Currently the frontend expects a valid body even in the error case. Make sure that is
    what the API returns.
    """
```

File: tests/api/test_store.py (hunk 1 above)
Comment (agners) [reply tests/api/test_store.py:199]: Agreed, added a suggestion.

File: tests/api/test_store.py [hunk 3]
```diff
@@ -188,3 +188,18 @@ async def container_events_task(*args, **kwargs):
...
+
+
+@pytest.mark.parametrize("resource", ["store/addons", "addons"])
+async def test_api_store_addons_no_changelog(
+    api_client: TestClient, coresys: CoreSys, store_addon: AddonStore, resource: str
+):
+    """Test /store/addons/{addon}/changelog REST API.
+    
+    Currently the frontend expects a valid body even in the error case. Make sure that is
+    what the API returns.
```
Comment (sairon) [inline tests/api/test_store.py:200]: Fix ruff:
```suggestion
    """Test /store/addons/{addon}/changelog REST API.

    Currently the frontend expects a valid body even in the error case. Make sure that is
    what the API returns.
```

**Pull request merged**